common/workunit/referencedfilelist.cpp

```diff
@@ -474,13 +488,213 @@ void ReferencedFile::resolve(const char *dstCluster, const char *srcCluster, IUs
+    }
+
+    getLFNDirectoryUsingBaseDir(folder, pLogicalPath, defaultFolder.str());
+
+    const char *n = pLogicalPath;
+    const char* p;
+    do {
+        p = strstr(n,"::");
+        if(p)
+            n = p+2;
+    } while(p);
+    title.append(n);
+    return true;
+}
+
+static StringBuffer& constructFileMask(const char* filename, StringBuffer& filemask)
+{
+    filemask.clear().append(filename).toLowerCase().append("._$P$_of_$N$");
+    return filemask;
+}
+
+static void setRoxieClusterPartDiskMapping(const char *clusterName, const char *defaultFolder, const char *defaultReplicateFolder, bool supercopy, IDFUfileSpec *wuFSpecDest, IDFUoptions *wuOptions)
+{
+    ClusterPartDiskMapSpec spec;
+    spec.setDefaultBaseDir(defaultFolder);
+
+    if (!supercopy)
+        spec.setRepeatedCopies(CPDMSRP_lastRepeated,false);
+    wuFSpecDest->setClusterPartDiskMapSpec(clusterName,spec);
+}
+
+//tbd make these parameters to requests using --dfu-copy
+static void getDefaultDFUName(StringBuffer &dfuQueueName)
+{
+// Using the first queue for now.
+#ifdef _CONTAINERIZED
+    Owned<IPropertyTreeIterator> dfuQueues = getComponentConfigSP()->getElements("dfuQueues");
+    ForEach(*dfuQueues)
+    {
+        const char *dfuName = dfuQueues->query().queryProp("@name");
+        if (!isEmptyString(dfuName))
+        {
+            getDfuQueueName(dfuQueueName, dfuName);
+            break;
+        }
+    }
+#else
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory(true);
+    Owned<IConstEnvironment> env = factory->openEnvironment();
+
+    StringBuffer xpath ("Software/DfuServerProcess");
+    Owned<IPropertyTree> root = &env->getPTree();
+    Owned<IPropertyTreeIterator> targets = root->getElements(xpath.str());
+    ForEach(*targets)
+    {
+        IPropertyTree &target = targets->query();
+        if (target.hasProp("@queue"))
+            dfuQueueName.set(target.queryProp("@queue"));
+    }
+#endif
+}
+
+
+static void dfuCopy(const char *dfu_queue, StringBuffer &publisherWuid, IUserDescriptor *user, const char *sourceLogicalName, const char *destLogicalName, const char *destNodeGroupIn, const char *srcDali, bool supercopy, bool overwrite, bool preserveCompression, bool nosplit)
+{
+    if(isEmptyString(publisherWuid))
+        throw MakeStringException(-1, "Failed to create Publisher DFU Workunit.");
+    if(isEmptyString(sourceLogicalName))
+        throw MakeStringException(-1, "Source logical file not specified.");
+    if(isEmptyString(destLogicalName))
+        throw MakeStringException(-1, "Destination logical file not specified.");
+
+    PROGLOG("Copy from %s to %s", sourceLogicalName, destLogicalName);
+
+    StringBuffer destFolder, destTitle, defaultFolder, defaultReplicateFolder;
+    StringBuffer srcNodeGroup;
+    StringBuffer destNodeGroup(destNodeGroupIn);
+    if(destNodeGroup.isEmpty())
+    {
+        getNodeGroupFromLFN(user, sourceLogicalName, destNodeGroup);
```

Comment:
This was how FileCopy was doing it, but I've decided not to try to guess.  The destination group/plane should have already been provided otherwise I will error.